networkit/cpp/distance/MultiTargetBFS.cpp

```diff
@@ -0,0 +1,66 @@
+#include <networkit/distance/BFS.hpp>
+#include <networkit/distance/MultiTargetBFS.hpp>
+
+#include <queue>
+#include <unordered_set>
+
+namespace NetworKit {
+
+void MultiTargetBFS::run() {
+    distances.clear();
+    distances.reserve(targets.size());
+    targetIdx.clear();
+
+    std::unordered_set<node> targetsSet(targets.begin(), targets.end());
+    count level = 0;
+
+    BFS bfs(*G, source);
+    bfs.run();
+    std::queue<node> frontier, next;
+    frontier.push(source);
+
+    std::vector<bool> visited(G->upperNodeIdBound());
+
+    auto visitNode = [&](node u) -> bool {
+        if (visited[u]) // Already visited, nothing to do
+            return false;
+
+        // Check if we are visiting a target node
+        const auto it = targetsSet.find(u);
+        if (it != targetsSet.end()) {
+            targetIdx.emplace(u, distances.size());
+            distances.push_back(static_cast<edgeweight>(level));
```

Comment:
I agree, but this might not be possible in C++ (e.g., when the targets are stored in an `unordered_set`).

Reply:
Right, I wasn't aware of `unordered_set`. In that case I would suggest to document the behavior for C++ (the need to iterate through `getTargetIndexMap()` to get correct target value). For the Python interface I think it is a better solution to return a correctly indexed distance list (since Cython mapping is `list` <-> `vector`).

Reply:
This is now implemented by keeping a copy of the targets list, which is used to sort the distances.